pywr/profiler.py

```diff
@@ -0,0 +1,59 @@
+import time
+import resource
+import sys
+from dataclasses import dataclass, asdict
+import pandas
+
+
+def memory_usage() -> float:
+    rusage_denom = 1024.0
+    if sys.platform == "darwin":
+        # ... it seems that in OSX the output is different units ...
+        rusage_denom = rusage_denom * rusage_denom
+    return resource.getrusage(resource.RUSAGE_SELF).ru_maxrss / rusage_denom
+
+
+@dataclass
+class ProfilerEntry:
+    phase: str
+    class_name: str
+    name: str
+    perf_counter: float
+    memory_usage: float
+
+
+class Profiler:
```

Comment:
@Batch21 do we want to keep this bundled with Pywr? Or should I remove it?

Reply:
I think it could be useful to keep but maybe with an expanded docstring that shows how to use it.

Reply:
OK, I'll have to make some changes because the `resources` module does not work on Windows.